policies/recipes/steps.go

```diff
@@ -484,13 +484,14 @@ func writeScript(path, contents string) error {
 func executeCommand(cmd string, args []string, workDir string, runEnvs []string, allowedExitCodes []int32) error {
 	cmdObj := exec.Command(cmd, args...)
 	cmdObj.Dir = workDir
+	cmdObj.Env = append(cmdObj.Env, os.Environ()...)
```

Comment:
for some reason starting powershell was failing when it was run as a service. Adding in os.Environ seemed to fix it (I think that Powershell is expecting certain variables to be setup when it starts)

Reply:
in general, we want a clean environment without any carried-over environment variables for working space. can we figure out what specifically powershell needs to be set?

Reply:
This would be very complicated, the environment powershell expects is a combination of variables that you should not try to create by hand. The varaibles you need a a combination of explicit system variables (HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment), and implicit variables like ProgramFiles,COMPUTERNAME,HOMEDRIVE,SystemDrive,SystemRoot...

I think using CreateEnvironmentBlock would be the 'correct' way if we wanted a clean environment, but here in our case I think just inheriting the parents makes sense and shouldn't cause us issues:
https://docs.microsoft.com/en-us/windows/win32/api/userenv/nf-userenv-createenvironmentblock?redirectedfrom=MSDN
https://godoc.org/golang.org/x/sys/windows#CreateEnvironmentBlock

some more reading:
https://docs.microsoft.com/en-us/windows/win32/procthread/environment-variables

Reply:
So the documents that Andrew posted matches up with what I found. @hopkiw how important is not carrying over variables for this? Generally our service will be running as a service on the machine and I expect the environment there should be essentially the same as if we created a new block. Also would it be worth it to just add these environment variables for powershell execution? My worry is that powershell won't be the only thing that requires an initialized environment.

Reply:
So I did some more digging and changed my mind slightly on this. I added methods to create a 'default' environment, which is created from the current process token on windows and just returns an empty slice on Linux.
I think creating the environment differently per OS should be fine and to me this seems like it gets a fairly clean environment without jumping through too many hoopy.

https://godoc.org/golang.org/x/sys/windows#GetCurrentProcessToken
https://godoc.org/golang.org/x/sys/windows#Token.Environ